src/Featurizers/CountVectorizerFeaturizer.h

```diff
@@ -0,0 +1,367 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/DocumentStatisticsEstimator.h"
+#include "../Traits.h"
+#include "../Strings.h"
+#include "Structs.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         CountVectorizerTransformer
+///  \brief         Returns SparseVectorEncoding<std::uint32_t> for each unique input
+///
+class CountVectorizerTransformer : public StandardTransformer<std::string, SparseVectorEncoding<std::uint32_t>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                           = StandardTransformer<std::string, SparseVectorEncoding<std::uint32_t>>;
+    using IndexMapType                       = std::unordered_map<std::string, std::uint32_t>;
+    using IterRangeType                      = std::tuple<std::string::const_iterator, std::string::const_iterator>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    IndexMapType const                       Labels;
+    bool const                               Binary;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    CountVectorizerTransformer(IndexMapType map, bool binary);
+    CountVectorizerTransformer(Archive &ar);
+
+    ~CountVectorizerTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(CountVectorizerTransformer);
+
+    void save(Archive &ar) const override;
+
+    bool operator==(CountVectorizerTransformer const &other) const;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methodsa
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC has problems when the definition and declaration are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+
+        using AppearanceMapType = typename std::map<IterRangeType, std::uint32_t, Components::IterRangeComp>;
+
+        AppearanceMapType appearanceMap;
+
+        Strings::Parse<std::string::const_iterator>(

```

Comment:
The parse method in execute() needs to be the same as fit() in DocumentStatisticEstimator, let's sync on this afternoon. The execute() in TfidfTransformer is a quick reference for now